src/fetch_data.js

```diff
@@ -28,41 +27,6 @@ export const getChangesetCoverageSummary = changeset =>
 export const getRawFile = (revision, path) =>
   fetch(`${hgHost}/integration/mozilla-inbound/raw-file/${revision}/${path}`, { plainHeaders });
 
-// Taken from https://github.com/mozilla/moz-coco/blob/master/src/client/Client.js
-// On October 23, 2017
-// Under the MPL License
-export const getFileRevisionCoverage = (revision, path, callback) => {
-  const body = Query.testCoverage(revision, path);
-  const jsonbody = JSON.stringify(body);
-  const options = {
-    hostname: `${activeData}`,
-    port: 443,
-    path: '/query',
-    method: 'POST',
-    headers: {
-      'Content-Type': 'text/plain',
-      'Content-Length': Buffer.byteLength(jsonbody)
-    }
-  }
-  console.log("Query sent: " + jsonbody);
-  const respchunks = [];
-  const p = new Promise((resolve, reject) => {
-    const req = https.request(options, (res) => {
-      res.setEncoding('utf8');
-      res.on('data', (chunk) => {
-        respchunks.push(new Buffer(chunk));
-      });
-      res.on('end', (chunk) => {
-        resolve(Buffer.concat(respchunks).toString('utf8'));
-      });
-      res.on('error', (e) => {
-        reject(e);
-      });
-    });
-    req.write(jsonbody);
-    req.end();
-  });
-  p.then((body) => {
-    callback(JSON.parse(body));
-  }).catch((err) => console.log(`Exception in fetch_data.js: ${err}`));
-}
\ No newline at end of file
+// get coverage from ActiveData for a particular source file
+export const getFileRevisionCoverage = (revision, path) =>
+  fetch(`${activeData}/query`, { jsonHeaders, method:"POST", body: JSON.stringify(Query.testCoverage(revision, path)) });
```

Comment:
Nice fix! I tried this but it didn't work for me for some reason.

Reply:
Yes, this is a nice line!